.github/workflows/ci.yml

```diff
@@ -18,18 +18,21 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v2
         with:
-          python-version: 2.7
+          python-version: 3.8
 
       - name: Install Python dependencies
         run: |
           pip install --upgrade pip
-          pip install yadage
+          pip install --pre reana-client
 
-      - name: Validate workflow
-        run: yadage-validate workflow/yadage/workflow.yaml | grep -q 'workflow validates'
+      - name: Validate workflows
+        run: |        
+          for file in $FILES; do reana-client validate -f $file; done
+        env:
+          FILES: reana.yaml reana-slurmcern.yaml reana-htcondorcern.yaml reana-yadage.yaml reana-yadage-slurmcern.yaml reana-yadage-htcondorcern.yaml reana-cwl.yaml reana-cwl-slurmcern.yaml reana-cwl-htcondorcern.yaml reana-snakemake.yaml
 
-      - run: rm -rf results && mkdir results
-
-      - run: docker run -i --rm -v `pwd`/code:/code -v `pwd`/data:/data -v `pwd`/results:/results python:2.7-slim python /code/helloworld.py --sleeptime 0
-
-      - run: grep -q 'Hello Joe Bloggs!' results/greetings.txt
+      - name: Run workflows
+        run: |
+          rm -rf results && mkdir results
+          docker run -i --rm -v `pwd`/code:/code -v `pwd`/data:/data -v `pwd`/results:/results python:2.7-slim python /code/helloworld.py --sleeptime 0
+          grep -q 'Hello Joe Bloggs!' results/greetings.txt
```

Comment:
This covers serial workflows, should we add steps to test Yadage, CWL and Snakemake locally in a similar way?

> @tiborsimko: If we want to get extra fancy, we could introduce separate "Run workflow" tests with running on naked CWL/Yadage/Serial-on-Docker/Snakemake environment, using the usual out-of-REANA instructions that we document there, i.e. mkdir snakemake-local-run etc.

Reply:
Could you please remind me where are those instructions? I'm sure I have seen them but I cannot find them

Reply:
- Yadage: https://github.com/reanahub/reana-demo-helloworld/blob/4599cdd365a59480abe6c1fc998c12f21f40ab5f/workflow/yadage/workflow.yaml#L1-L13
- CWL: https://github.com/reanahub/reana-demo-helloworld/blob/4599cdd365a59480abe6c1fc998c12f21f40ab5f/workflow/cwl/helloworld.cwl#L1-L12
- Snakemake: https://github.com/reanahub/reana-demo-helloworld/blob/4599cdd365a59480abe6c1fc998c12f21f40ab5f/workflow/snakemake/Snakefile#L1-L14